webapp/Connector/src/app/store/StudyProducts.js

```diff
@@ -94,6 +103,22 @@ Ext.define('Connector.app.store.StudyProducts', {
         this._onLoadComplete();
     },
 
+    getById : function(id) {
```

Comment:
Can we simply push all products to product store, but use a flag showSummary to decide if the record should be shown. It would be much cleaner to have everything in store.